python/cudf/cudf/core/reshape.py

```diff
@@ -384,11 +382,10 @@ def length_check(obj, name):
                 prefix_sep=prefix_sep_map.get(name, prefix_sep),
                 dtype=dtype,
             )
-            df_list.append(col_enc_df)
+            for col in col_enc_df.columns.difference(df._data.names):
+                result_df[col] = col_enc_df._data[col]
```

Comment:
Should this be outside of the `for name in columns` loop?

Reply:
This can be inside for loop as we are actually operating on `col_enc_df`.